### DISCUSSION TITLE
Why RabbitMQ does not support downgrade/rollback?

### DISCUSSION THREAD

**Discussion by manoj365study** [rabbitmq/rabbitmq-server#12075] [Questions] [answered]
As Rabbitmq documentation page says "RabbitMQ does not support downgrades; it's strongly advised to back node's data directory up before upgrading." https://www.rabbitmq.com/docs/upgrade#single-node-upgrade

It is not said that why it is not supported and challenges to support rollback. 

Can Someone explain here, why RabbitMQ not supporting this feature. Would like to know more about it.

**Comment by michaelklishin** [comment, marked as answer]
Because certain data migrations are irreversible and others, while possible, would require a substantial effort for our small team.

In many cases you would be able go back to a different version using a [Blue-Green Deployment](https://www.rabbitmq.com/docs/upgrade) upgrade. Unless you adopt any of the features (not to be confused with feature flags) of the new version which would fail to import into the Green cluster on an older version.
